src/Sulu/Component/Content/Document/Query/SuluNodeInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Document\Query;
+
+use Doctrine\ODM\PHPCR\Query\Builder\AbstractNode;
+
+/**
+ * All Sulu query builder nodes implement this interface.
+ */
+interface SuluNodeInterface
+{
+    /**
+     * Return a the Doctrine PHPCR-ODM instance of the node.
+     *
+     * See the documentation in the Sulu\Component\Content\Document\Query\StructureQueryBuilderConverter
+     * for information on why this is necessary.
```

Comment:
No, it is necessary because the PHPCR-ODM query converter will not accept our node classes. The documentation your referring to is in `QueryBuilder` class.